cedar-policy/src/api.rs

```diff
@@ -1680,54 +1645,66 @@ impl FromStr for PolicySet {
 
 impl PolicySet {
     /// Build the policy set AST from the EST
-    fn from_est(est: est::PolicySet) -> Result<Self, PolicySetError> {
-        let mut pset = Self::default();
-
-        for PolicyEntry { id, policy } in est.templates {
-            let template = Template::from_est(Some(PolicyId::new(id)), policy)?;
-            pset.add_template(template)?;
-        }
-
-        for PolicyEntry { id, policy } in est.static_policies {
-            let p = Policy::from_est(Some(PolicyId::new(id)), policy)?;
-            pset.add(p)?;
-        }
-
-        for Link {
-            id,
-            template,
-            slots,
-        } in est.links
-        {
-            let slots = slots
-                .into_iter()
-                .map(|(key, value)| (key.into(), EntityUid::new(value)))
-                .collect();
-            pset.link(PolicyId::new(template), PolicyId::new(id), slots)?;
-        }
-
-        Ok(pset)
+    fn from_est(est: &est::PolicySet) -> Result<Self, PolicySetError> {
+        let ast: ast::PolicySet = est.clone().try_into()?;
```

Comment:
If we're going to unconditionally `.clone()` the `est` in this function, I'd rather take an owned `est` (like the current code); callers can call `.clone()` if they need the clone, but if they don't need ownership after this call, we avoid the clone.  (And looking at the callers, in this case we can indeed avoid the clone)

Reply:
I'd still take an owned argument, and consume it at the last time it's used in this function (i.e., when we're constructing the lossless policy).  Or is does its last use only require a reference?

Reply:
Constructing the lossless policies only requires a reference. (Although behind the scenes every policy ends up cloned via `.get_template()` or `.get_policy()`.)

Reply:
Ick. Ok, probably nothing to do in this PR; if the cloning becomes a performance problem, we'll address it then.